addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/ClassicWiFiLEDDriver.java

```diff
@@ -149,16 +170,28 @@ private void sendLEDData(LEDStateDTO ledState) throws IOException {
             byte w = (byte) (((int) (ledState.getWhite().doubleValue() * 255 / 100)) & 0xFF);
             byte w2 = (byte) (((int) (ledState.getWhite2().doubleValue() * 255 / 100)) & 0xFF);
 
-            byte[] bytes = getBytesForColor(r, g, b, w, w2);
+            bytes = getBytesForColor(r, g, b, w, w2);
 
-            sendRaw(bytes);
         } else {
             // program selected
             byte p = (byte) (program & 0xFF);
             byte s = (byte) (((100 - ledState.getProgramSpeed().intValue()) * 0x1F / 100) & 0xFF);
-            byte[] data = { 0x61, p, s };
-            sendRaw(data);
+            bytes = new byte[]{ 0x61, p, s };
         }
+
+        ledUpdateFuture = updateScheduler.submit(() -> {
+            try {
+                Thread.sleep(WAIT_UPDATE_LED_FOR_MS);
+                logger.debug("Setting LED State to {}", bytesToHex(bytes));
+                sendRaw(bytes);
+                return true;
+            } catch (IOException e) {
+                logger.warn("IOException", e);
```

Comment:
Can you take a look at the current situation, I don't know this piece of code and have no IDE at hand.